### PR TITLE
Fix `percentiles_summary` with `dask_cudf`

### PR DESCRIPTION
**Pull request by pentschev**
- [ ] Fixes issue introduced in #7305 for dask_cudf, as mentioned in https://github.com/dask/dask/pull/7305#issuecomment-791132948
- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/dataframe/partitionquantiles.py
Comment (quasiben) [inline dask/dataframe/partitionquantiles.py:417]: Thanks @pentschev! Do you think it's worth adding a comment saying that "nearest" option is not supported for cupy here ?

Comment (pentschev) [reply dask/dataframe/partitionquantiles.py:417]: Done in https://github.com/dask/dask/pull/7325/commits/012427342c0bc452675adbde6ebd47dfa18107b4

**Comment by quasiben** [comment]
Thanks again @pentschev .  I'll merge when tests pass

File: dask/dataframe/partitionquantiles.py
Comment (rjzamora) [reply dask/dataframe/partitionquantiles.py:417]: Thanks Peter!

The Dask PR removed the following rounding step after the `_percentiles` call:
```python
    if interpolation == "linear" and np.issubdtype(data.dtype, np.integer):
        vals = np.round(vals).astype(data.dtype)
```

Do we still need this if the data is cupy?

Comment (pentschev) [reply dask/dataframe/partitionquantiles.py:417]: I'm not sure about it, what I can say is that `dask_cudf` tests that were failing because of that in tonight's tests (see https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-gpu-test/CUDA=10.1,GPU_LABEL=gpu,OS=ubuntu16.04,PYTHON=3.7/1184/#showFailuresLink) pass for me locally. I'm happy to add that back if someone is confident on the correct usage here.

Comment (rjzamora) [reply dask/dataframe/partitionquantiles.py:417]: Got it - Makes perfect sense to use CI as a guide.  My worry is that we may run into other problems if we were previously ensuring that divisions were the same dtype as the data, and now it will always be float.

Maybe, while we are already making a change, we can fix the origial issue for cupy data, and set the 0 and 100 percentiles to the actual min/max values?

**Comment by jakirkham** [comment]
cc @jrbourbeau

**Comment by quasiben** [comment]
Thanks for the reminder @jakirkham .  I'll wait for @jrbourbeau to sign-off (james, you can also merge if you are good with this)

**Review by jakirkham — APPROVED** [review]

File: dask/dataframe/partitionquantiles.py
Comment (rjzamora) [reply dask/dataframe/partitionquantiles.py:417]: To follow up on my previous comment.  It may be best to do this after the `_percentiles` call:

```python
if is_cupy_type(data) and interpolation == "linear" and np.issubdtype(data.dtype, np.integer):
    vals = np.round(vals).astype(data.dtype)
    if qs[0] == 0:
        # Ensure the 0th quantile is the minimum value of the data
        vals[0] = data.min()
```

Comment (pentschev) [reply dask/dataframe/partitionquantiles.py:417]: Thanks @rjzamora for the suggestion, I can confirm locally `dask_cudf` tests also pass with the change above, so I think it's probably good to do it. I pushed https://github.com/dask/dask/pull/7325/commits/3a9fdf5d0cdd8472b23ca86201099e98d110a35e with the changes you suggested now.

**Comment by jakirkham** [comment]
Looks like we need to run `black`

**Review by kylebarron — APPROVED** [review]

**Comment by jrbourbeau** [comment]
I took the liberty of running `black` (hope you don't mind Peter)

**Comment by pentschev** [comment]
Not at all, thanks @jrbourbeau !

**Review by jakirkham — APPROVED** [review]

**Review by rjzamora — APPROVED** [review]

**Review by jrbourbeau — APPROVED** [review]
Thanks all! Let's merge after CI passes

File: dask/dataframe/partitionquantiles.py
Comment (kylebarron) [inline dask/dataframe/partitionquantiles.py:405]: Any reason not to have this import at the top level?

Comment (pentschev) [reply dask/dataframe/partitionquantiles.py]: No special reason here, I moved it to the top now and hopefully CI will pass this time, although it seems the failure was some sort of timeout https://github.com/dask/dask/pull/7325/checks?check_run_id=2041407897 .

**Comment by jakirkham** [comment]
Thanks Peter! Also thanks everyone for the reviews! 😄

**Pull request merged**
